### PR TITLE
API improvements for BPF LSM programs

### PR DESCRIPTION
**Pull request by willfindlay**
This patch improves the API for BPF LSM programs to bring it in line with other program types.
Specifically, it adds helpers for attaching and detaching LSM programs, and adds calls to these helpers in `__trace_autoload` so that LSM programs are automatically attached when they have the correct prefix. Finally, it introduces a `support_lsm` method that indicates kernel support for BPF LSM programs.

### CODE REVIEW

**Comment by yonghong-song** [comment]
[buildbot, test this please]

File: src/cc/libbpf.c [hunk 1]
```diff
@@ -1203,6 +1203,23 @@ int bpf_attach_kfunc(int prog_fd)
   return ret;
 }
 
+int bpf_detach_lsm(int prog_fd, char *func)
```
Comment (yonghong-song) [inline src/cc/libbpf.c:1206]: could you remove bpf_detach_lsm? There is no need from python side as python code will close the fd. We can add it later if it is truely used in C++ side.

Comment (willfindlay) [reply src/cc/libbpf.c:1206]: @yonghong-song Yes I can remove. But I was copying what was done for kfunc. Perhaps it should be removed for kfunc as well?

Comment (yonghong-song) [reply src/cc/libbpf.c:1206]: Yes, let us remove it as well. The functions are not used. Let us add them if there is a need in C++ side later. Thanks!

Comment (willfindlay) [reply src/cc/libbpf.c]: @yonghong-song I have now removed both detach_lsm and detach_kfunc

**Comment by yonghong-song** [comment]
[buildbot, test this please]

**Comment by yonghong-song** [comment]
@willfindlay could you rebase your patch with latest mater? The github complains it cannot be rebased safely in order to merge. Thanks!

**Comment by yonghong-song** [comment]
[buildbot, ok to test]

**Comment by willfindlay** [comment]
@yonghong-song Hopefully it is okay now

**Pull request merged**